### PR TITLE
lis2dw12: Add support for set/clear int cfg bits

### PR DESCRIPTION
**Pull request by vrahane**
- Set/Clear individual interrupt cfg bits
- Change config in sensor creator
- Change int_active to int_active_low to be more explicit
- Change lis2dw12_set_int_active() to lis2dw12_set_int_active_low()

### CODE REVIEW

File: hw/drivers/sensors/lis2dw12/src/lis2dw12.c [hunk 1]
```diff
@@ -1949,92 +2034,93 @@ init_intpin(struct lis2dw12 *lis2dw12, hal_gpio_irq_handler_t handler,
...
     pdd = &lis2dw12->pdd;
 
-    rc = lis2dw12_clear_int(itf, &reg);
-    if (rc) {
-        goto err;
-    }
-
-    /* if no interrupts are currently in use enable int pin */
+    /* disable int pin */
     if (!pdd->int_enable) {
```
Comment (ccollins476ad) [inline hw/drivers/sensors/lis2dw12/src/lis2dw12.c]: Should this be:
```
if (pdd->int_enable == 1) {
```
?

File: hw/drivers/sensors/lis2dw12/include/lis2dw12/lis2dw12.h [hunk 2]
```diff
@@ -579,6 +579,26 @@ lis2dw12_set_int1_pin_cfg(struct sensor_itf *itf, uint8_t cfg);
  */
 int lis2dw12_set_int2_pin_cfg(struct sensor_itf *itf, uint8_t cfg);
 
+/**
+ * Clear interrupt pin configuration for interrupt 1
+ *
+ * @param the sensor interface
```
Comment (ccollins476ad) [inline hw/drivers/sensors/lis2dw12/include/lis2dw12/lis2dw12.h]: `itf` is missing from this `@param` comment (same for `lis2dw12_clear_int1_pin_cfg`).

**Comment by ccollins476ad** [comment]
I know I already mentioned this, but I get a little nervous about tracking the count of enabled interrupts without knowing which ones are actually enabled.  I don't know this sensor code very well, so I may well be worrying over something that doesn't actually matter.  I would just be sure that it isn't possible to disable the same interrupt twice, and/or disable interrupts too many times and cause the interrupt count to roll over.  There, I said my piece :).

File: hw/drivers/sensors/lis2dw12/src/lis2dw12.c (hunk 1 above)
Comment (vrahane) [reply hw/drivers/sensors/lis2dw12/src/lis2dw12.c]: Taking care of this.

File: hw/drivers/sensors/lis2dw12/include/lis2dw12/lis2dw12.h (hunk 2 above)
Comment (vrahane) [reply hw/drivers/sensors/lis2dw12/include/lis2dw12/lis2dw12.h]: Do you mean the `itf` text in the comments ? None of my comments have that. Is it needed ?

**Comment by vrahane** [comment]
@ccollins476ad changed all the comments to match doxygen format. Please check it out.

**Pull request merged**